### PR TITLE
added python docs for classes in optimizer.py, removed a redundant st…

### PR DESCRIPTION
**Pull request by shane-huang**
…ate config batchSize in example, scala doc for Trigger, Poly and Step in SGD

Mostly documentation change

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/625

### CODE REVIEW

**Comment by zhichao-li** [comment]
LGTM. Let's merge it once pass the style check.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Trigger.scala [hunk 1]
```diff
@@ -60,6 +82,11 @@ object Trigger {
     }
   }
 
+  /**
+   * A trigger that triggers an action when training reaches
+   * the number of iterations specified by "max".
+   * Usually used in Optimizer.setEndWhen.
+   */
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Trigger.scala]: `@param max`

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Trigger.scala [hunk 2]
```diff
@@ -52,6 +69,11 @@ object Trigger {
     }
   }
 
+  /**
+   * A trigger that triggers an action when training reaches
+   * the number of epochs specified by "max".
+   * Usually used in Optimizer.setEndWhen.
+   */
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Trigger.scala]: `@param max`

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/SGD.scala [hunk 3]
```diff
@@ -148,6 +148,12 @@ object SGD {
       config("clr") = -config.get[Double]("learningRate").getOrElse(1e-3)
     }
   }
+
+  /**
+   * A learning rate decay policy, where the effective learning rate
+   * follows a polynomial decay, to be zero by the max_iteration.
+   * Calculation: base_lr (1 - iter/maxIteration) ^ (power)
+   */
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/SGD.scala]: `@param`

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/SGD.scala [hunk 4]
```diff
@@ -162,7 +168,10 @@ object SGD {
       config("clr") = clr
     }
   }
-
+  /**
+   * A learning rate decay policy, where the effective learning rate
+   * is calculated as base_lr * gamma ^ (floor(iter / stepSize))
+   */
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/SGD.scala]: `@param`

**Pull request closed without merging**